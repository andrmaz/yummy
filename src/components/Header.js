import React from 'react';
import { Button, useColorMode, Icon, Heading } from '@chakra-ui/react';
import { FaUserCircle } from 'react-icons/fa';

function Header() {

    const { colorMode, toggleColorMode } = useColorMode();
    
    return (
        <Heading d='flex' justifyContent='space-between' pt='10px'>
          {/* Switch light and dark mode */}
          <Button data-testid="toggle" onClick={toggleColorMode} >
            Toggle {colorMode === 'light' ? 'Dark' : 'Light'}
          </Button>
          <Icon as={FaUserCircle} />
        </Heading>
    );
};

export default Header;
