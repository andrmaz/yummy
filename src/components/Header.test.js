import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';

let wrapper;

describe('Header Component', () => {
    beforeEach(() => {
        wrapper = shallow(<Header />);
        expect.hasAssertions();
    });
    it('should render Component correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
    it('should display Toggle Light at first', () => {
        expect(wrapper.text()).toEqual('Toggle Light');
    });
});
