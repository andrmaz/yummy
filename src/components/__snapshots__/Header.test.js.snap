// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should render Component correctly 1`] = `
<Heading
  d="flex"
  justifyContent="space-between"
  pt="10px"
>
  <Button
    data-testid="toggle"
  >
    Toggle 
    Light
  </Button>
  <Icon
    as={[Function]}
  />
</Heading>
`;
